plugins/modules/ec2_asg_instance_refresh.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: ec2_asg_instance_refresh
+version_added: 1.0.0
+short_description: Start or cancel an  ec2 Auto Scaling Group Instance Refresh in AWS
+description:
+  - Start or cancel an ec2 Auto Scaling Group Instance Refresh in AWS
+  - Can be used with ec2_asg_instance_refreshes_info to track the subsequent progress
+requirements: [ boto3 ]
+author: "Dan Khersonsky (@danquixote)"
+options:
+  state:
+    description:
+      - Desired state
+    type: str
+    required: true
+    choices: [ 'started', 'canceled' ]
+  name:
+    description:
+      - The name of the auto scaling group you are searching for.
+    type: str
+    required: true
+  strategy:
+    description:
+      - list of InstanceRefreshIds
+    type: str
+    default: 'Rolling'
+  preferences:
+    description:
+      - preferences
+    required: false
+    suboptions:
+      min_healthy_percentage:
+        description:
+          - min_healthy_percentage
+        type: int
+      instance_warmup:
+        description:
+          - instance_warmup
+        type: int
+    type: dict
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Start a refresh
+  community.aws.ec2_asg_instance_refresh:
+    name: some-asg
+    state: started
+
+- name: Cancel a refresh
+  community.aws.ec2_asg_instance_refresh:
+    name: some-asg
+    state: canceled
+
+- name: Start a refresh and pass preferences
+  community.aws.ec2_asg_instance_refresh:
+    name: some-asg
+    state: started
+    preferences:
+      min_healthy_percentage: 91
+      instance_warmup: 60
+
+'''
+
+RETURN = '''
+---
+instance_refresh_id:
+    description: instance refresh id
+    returned: success
+    type: str
+    sample: "08b91cf7-8fa6-48af-b6a6-d227f40f1b9b"
+'''
+
+import re
+import sys
+
```

Comment:
```suggestion
```

I don't see either of these imports being used.

Reply:
removed in 7b7ea5bebf0c69a8d8e7f6e1b30c3f7f166427c5